Domains/4-Application/ISM/Released/IntegratedStructuralModel.02.00.00.ecschema.xml

```diff
@@ -0,0 +1,2764 @@
+        <ECProperty propertyName="AutoDetailedLength" typeName="double" displayLabel="Detailed Length" category="Calculated_category" kindOfQuantity="AECU:LENGTH" description="Auto-calculated detailed bar length (including hooks)." />
+        <ECProperty propertyName="AutoGroupBarWeight" typeName="double" displayLabel="Group Bar Weight" category="Calculated_category" kindOfQuantity="AECU:WEIGHT" description="Auto-calculated detailed bar weight (including hooks)." />
+        <ECProperty propertyName="BarCount" typeName="int" displayLabel="Bar Count" category="IsmConcentratedSurfaceRebar_category" description="The number of bars." />
+        <ECProperty propertyName="LayoutWidth" typeName="double" displayLabel="Layout Width" category="IsmConcentratedSurfaceRebar_category" kindOfQuantity="AECU:LENGTH" description="Rebar Layout Width." />
+        <ECProperty propertyName="ConstrainedDimension" typeName="ConstrainedDimensionType" category="IsmConcentratedSurfaceRebar_category" displayLabel="Constrained Dimension" description="Defines which property is auto-calculated." />
+        <ECProperty propertyName="StaggerLength" typeName="double" displayLabel="Stagger Length" category="IsmConcentratedSurfaceRebar_category" kindOfQuantity="AECU:LENGTH" description="The length of the bar that alternate bars are staggered by." />
+    </ECEntityClass>
+
+    <PropertyCategory typeName="IsmAreaSurfaceRebar_category" displayLabel="Area Surface Rebar" priority="170" />
+    <ECEntityClass typeName="IsmAreaSurfaceRebar" modifier="Sealed" displayLabel="Area Surface Rebar">
+        <BaseClass>IsmSurfaceRebar</BaseClass>
+        <ECStructProperty propertyName="LayoutBoundary" typeName="IsmPrimitiveGeometry" displayLabel="Layout Boundary" category="IsmAreaSurfaceRebar_category" description="The planar boundary of the region filled with bars." />
+        <ECProperty propertyName="ConsiderMemberOpenings" typeName="boolean" displayLabel="Consider Member Openings" category="IsmAreaSurfaceRebar_category" description="Should rebars be placed within openings?" />
+        <ECProperty propertyName="EndAndSideCover" typeName="double" displayLabel="End and Side Cover" category="IsmAreaSurfaceRebar_category" kindOfQuantity="AECU:LENGTH_SHORT" description="An inset cover." />
+        <ECProperty propertyName="AutoBarCount" typeName="int" displayLabel="Bar Count" category="Calculated_category" description="Auto-calculated total bar count." />
+        <ECProperty propertyName="AutoGroupBarWeight" typeName="double" displayLabel="Group Bar Weight" category="Calculated_category" kindOfQuantity="AECU:WEIGHT" description="Auto-calculated total group bar weight." />
+    </ECEntityClass>
+
+    <PropertyCategory typeName="IsmGrid_category" displayLabel="Grid" priority="190" />
+    <ECEntityClass typeName="IsmGrid" modifier="Abstract" displayLabel="Grid" >
+        <BaseClass>bis:PhysicalElement</BaseClass>
```

Comment:
Even though this version of the ISM BIS schema matches more closely the EC2 version of ISM and therefore it embraces very little of the core rules of BIS, I'm concerned that it doesn't at least try to distinguish between physical and spatial-location concepts. e.g. concepts such as "Grid" or "GridLine" are clearly spatial-location, not physical, yet their base-classes are bis:PhysicalElement.

There are downstream consumers that rely on the correct classification between physical and non-physical (e.g. clash-detection).

Would it be possible to make such a change in base classes for the ISM concepts that are currently subclasses of bis:PhysicalElement but are clearly spatial-location?

Reply:
The aforementioned PR merged before I could review it. I just noticed that the new base-class of those concepts was changed to bis:SpatialElement. Such base-class is not supposed to be subclassed outside of BisCore - unfortunately it got released without the custom-attribute that prevents that from happening back in the day. We haven't addressed that due to fear of breaking older iModels (cc: @ColinKerr ).

With that said, the correct base-class in the cases above is bis:SpatialLocationElement - not bis:SpatialElement.

Reply:
Did not notice this comment. Will make new releases.

Reply:
I've merged changes to this PR